### PR TITLE
READY Update simnet tests to v5

### PR DESCRIPTION
**Pull request by radupopa2010**
This PR updates simnet tests to v5 version.

After discussion with @TriplEight  I edited this PR to update to simnet v5.

trigger_pipeline.sh was copied  from  https://github.com/paritytech/polkadot/blob/master/scripts/gitlab/trigger_pipeline.sh
and adapted.

We reviewed together https://github.com/paritytech/polkadot/blob/master/scripts/gitlab/trigger_pipeline.sh
last week 

Pipeline is green.
https://gitlab.parity.io/parity/substrate/-/pipelines/141040

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -696,6 +705,7 @@ trigger-simnet:
     - if: $CI_PIPELINE_SOURCE == "pipeline"
       when: never
     - if: $CI_PIPELINE_SOURCE == "web" && $CI_COMMIT_REF_NAME == "master"
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
```
Comment (TriplEight) [inline .gitlab-ci.yml:708]: we did not agree to run simnet from substrate on PRs, did it change somewhere?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -567,9 +569,16 @@ build-rust-doc:
...
+    - echo "IMAGE_NAME is ${IMAGE_NAME}"
     - echo "IMAGE_NAME=${IMAGE_NAME}" > ./artifacts/$PRODUCT/build.env
-    - echo "IMAGE_TAG=${VERSION}" >> ./artifacts/$PRODUCT/build.env
+    - ls -la
+    - VERSION="$(cat ./artifacts/$PRODUCT/VERSION)"
+    - echo "VERSION is ${VERSION}"
+    - IMAGE_TAG="${VERSION}"
+    - echo "IMAGE_TAG is ${IMAGE_TAG}"
+    - echo "IMAGE_TAG=${IMAGE_TAG}" >> ./artifacts/$PRODUCT/build.env
+    - cat ./artifacts/$PRODUCT/build.env
```
Comment (TriplEight) [inline .gitlab-ci.yml:581]: please remove the unneeded debug info like `ls -la` and echoing the same value 2 times.

**Review by TriplEight — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (radupopa2010) [reply .gitlab-ci.yml:708]: I think the earlier we catch a bug the better ; let's discuss this in simnet team channel

Comment (radupopa2010) [reply .gitlab-ci.yml:708]: maybe replace with
`    - if: $CI_COMMIT_MESSAGE  =~ /.*build-for-simnet.*/`

Comment (TriplEight) [reply .gitlab-ci.yml:708]: let's keep it on "revert me" basis for now

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -114,6 +114,7 @@ default:
     - if: $CI_PIPELINE_SOURCE == "schedule"
     - if: $CI_COMMIT_REF_NAME == "master"
     - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
+    - if: $CI_COMMIT_MESSAGE  =~ /.*build-for-simnet.*/
```
Comment (rcny) [inline .gitlab-ci.yml:117]: Looks like this line should be removed before the merge too. Or is it exist for any additional non-debug purposes?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -567,9 +569,14 @@ build-rust-doc:
     - buildah push --format=v2s2 "$IMAGE_NAME:latest"
   after_script:
     - buildah logout "$IMAGE_NAME"
-    # pass artifacts to the trigger-simnet job
+    # pass artifacts to the trigger-simnet job 
+    - echo "IMAGE_NAME is ${IMAGE_NAME}"
     - echo "IMAGE_NAME=${IMAGE_NAME}" > ./artifacts/$PRODUCT/build.env
```
Comment (rcny) [inline .gitlab-ci.yml:574]: We can use `tee -a` to get rid of double `echo` here but that's a nitpick, nvm.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -567,9 +569,14 @@ build-rust-doc:
...
-    - echo "IMAGE_TAG=${VERSION}" >> ./artifacts/$PRODUCT/build.env
+    - VERSION="$(cat ./artifacts/$PRODUCT/VERSION)"
+    - IMAGE_TAG="${VERSION}"
+    - echo "IMAGE_TAG is ${IMAGE_TAG}"
+    - echo "IMAGE_TAG=${IMAGE_TAG}" >> ./artifacts/$PRODUCT/build.env
```
Comment (rcny) [inline .gitlab-ci.yml:578]: Same as above

File: .gitlab-ci.yml (hunk 3 above)
Comment (radupopa2010) [reply .gitlab-ci.yml:117]: Agree, I can add a  "revert me" line here like

    `- if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/ # PRs  FIXME: delete me`

File: .gitlab-ci.yml (hunk 4 above)
Comment (radupopa2010) [reply .gitlab-ci.yml:574]: good tip, I will use it :)

**Review by rcny — APPROVED** [review]

**Comment by rcny** [comment]
No major CI changes, LGTM.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -708,4 +711,4 @@ trigger-simnet:
     DWNSTRM_ID:                    332
   script:
     # API trigger for a simnet job
-    - .maintain/gitlab/trigger_pipeline.sh
+    - .maintain/gitlab/trigger_pipeline.sh --simnet-version=${SIMNET_REF}
```
Comment (bkchr) [inline .gitlab-ci.yml:714]: Where is this `SIMNET_REF` defined?

**Review by bkchr — APPROVED** [review]

File: .gitlab-ci.yml (hunk 6 above)
Comment (radupopa2010) [reply .gitlab-ci.yml:714]: This is defined as a variable in gitlab https://gitlab.parity.io/parity/substrate/-/settings/ci_cd

**Review by rcny — APPROVED** [review]

**Comment by rcny** [comment]
bot merge

File: .gitlab-ci.yml (hunk 2 above)
Comment (TriplEight) [reply .gitlab-ci.yml:575]: @radupopa2010 piped `tee` does literally the same what `cat` does, why did you leave it here?

**Pull request merged**